internal/pkg/cli/deploy/workload_test.go

```diff
@@ -0,0 +1,1176 @@
+			inRegion:  "us-west-2",
+			mock: func(t *testing.T, m *deployMocks) {
+				m.mockFileReader.EXPECT().ReadFile(filepath.Join(mockWorkspacePath, mockEnvFile)).Return([]byte{}, nil)
+				m.mockUploader.EXPECT().Upload(mockS3Bucket, mockEnvFilePath, gomock.Any()).
+					Return(mockEnvFileS3URL, nil)
+				m.mockAddons.EXPECT().Package(gomock.Any()).Return(nil)
+				m.mockAddons.EXPECT().Template().Return("some data", nil)
+				m.mockUploader.EXPECT().Upload(mockS3Bucket, mockAddonPath, gomock.Any()).
+					Return(mockAddonsS3URL, nil)
+			},
+
+			wantAddonsURL:  mockAddonsS3URL,
+			wantEnvFileARN: mockEnvFileS3ARN,
+		},
+		"should return error if fail to upload to S3 bucket": {
+			inRegion: "us-west-2",
+			mock: func(t *testing.T, m *deployMocks) {
+				m.mockAddons.EXPECT().Package(gomock.Any()).Return(nil)
+				m.mockAddons.EXPECT().Template().Return("some data", nil)
+				m.mockUploader.EXPECT().Upload(mockS3Bucket, mockAddonPath, gomock.Any()).
+					Return("", mockError)
+			},
+
+			wantErr: fmt.Errorf("put addons artifact to bucket mockBucket: some error"),
+		},
+		"should return empty url if the service doesn't have any addons and env files": {
+			mock: func(t *testing.T, m *deployMocks) {
+				m.mockAddons = nil
+			},
+		},
+		"should fail if packaging addons fails": {
+			mock: func(t *testing.T, m *deployMocks) {
+				m.mockAddons.EXPECT().Package(gomock.Any()).Return(mockError)
+			},
+			wantErr: fmt.Errorf("package addons: %w", mockError),
+		},
+		"should fail if addons template can't be created": {
+			mock: func(t *testing.T, m *deployMocks) {
+				m.mockAddons.EXPECT().Package(gomock.Any()).Return(nil)
+				m.mockAddons.EXPECT().Template().Return("", mockError)
+			},
+			wantErr: fmt.Errorf("retrieve addons template: %w", mockError),
+		},
+	}
+
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
+
+			m := &deployMocks{
+				mockUploader:           mocks.NewMockuploader(ctrl),
+				mockAddons:             mocks.NewMockstackBuilder(ctrl),
+				mockImageBuilderPusher: mocks.NewMockimageBuilderPusher(ctrl),
+				mockFileReader:         mocks.NewMockfileReader(ctrl),
+			}
+			tc.mock(t, m)
+
+			wkldDeployer := &workloadDeployer{
+				name: mockName,
+				env: &config.Environment{
+					Name:   mockEnvName,
+					Region: tc.inRegion,
+				},
+				app: &config.Application{
+					Name: mockAppName,
+				},
+				resources:     mockResources,
+				imageTag:      mockImageTag,
+				workspacePath: mockWorkspacePath,
+				mft: &mockWorkloadMft{
+					fileName:      tc.inEnvFile,
+					buildRequired: tc.inBuildRequired,
+				},
+				fs:                 m.mockFileReader,
+				s3Client:           m.mockUploader,
+				imageBuilderPusher: m.mockImageBuilderPusher,
+				templateFS:         fakeTemplateFS(),
+			}
+			if m.mockAddons != nil {
+				wkldDeployer.addons = m.mockAddons
+			}
+			var deployer artifactsUploader
+			deployer = &lbWebSvcDeployer{
```

Comment:
It feels to me that the unit tests here in `workload_test.go` should be testing constructing a workloadDeployer and calls its `uploadArtifacts` method for testing. But I guess it doesn't matter to much given that lbws's `UploadArtifacts` is virtually workloadDeployer's `uploadArtifacts`